src/bridge/cockpitcpusamples.c

```diff
@@ -71,28 +73,45 @@ cockpit_cpu_samples (CockpitSamples *samples)
       guint64 system;
       guint64 idle;
       guint64 iowait;
+      gchar cpu_core[CPU_CORE_MAXLEN + 1];
+      gchar core[CPU_CORE_MAXLEN - 3 + 1]; // -3 as "cpu" is omitted
 
-      if (!(g_str_has_prefix (line, "cpu ")))
+      if (!(g_str_has_prefix (line, "cpu")))
         continue;
 
       #define FMT64 "%" G_GUINT64_FORMAT " "
-      if (sscanf (line + sizeof ("cpu ") - 1, FMT64 FMT64 FMT64 FMT64 FMT64,
+      if (sscanf (line, "%" G_STRINGIFY (CPU_CORE_MAXLEN) "s " FMT64 FMT64 FMT64 FMT64 FMT64,
+                  cpu_core,
                   &user,
                   &nice,
                   &system,
                   &idle,
-                  &iowait) != 5)
+                  &iowait) != 6)
         {
           g_warning ("Error parsing line %d of /proc/stat with content `%s'", n, line);
           continue;
         }
 
       user_hz = ensure_user_hz ();
-      cockpit_samples_sample (samples, "cpu.basic.nice", NULL, nice*1000/user_hz);
-      cockpit_samples_sample (samples, "cpu.basic.user", NULL, user*1000/user_hz);
-      cockpit_samples_sample (samples, "cpu.basic.system", NULL, system*1000/user_hz);
-      cockpit_samples_sample (samples, "cpu.basic.iowait", NULL, iowait*1000/user_hz);
-      break;
+      if (strlen (cpu_core) > 3)
+        {
+          if (sscanf (cpu_core, "cpu%s", core) != 1)
```

Comment:
Oh, just now I realized that this doesn't parse it into an integer, but a string. If that was by intent (indexing the series by string names instead of ints), then we don't need that sscanf() at all -- just `core+3` is fine to chop off the "cpu" prefix.

Reply:
We need string as later on the name must be string and not int. And indeed, `+3` would do, I was playing too much with `sscanf` that I wanted to use it for everything :)